### PR TITLE
Use Arcade's pool-providers.yml

### PR DESCRIPTION
**Pull request by dougbu**
- import the variable template
- use the two variables it provides wherever we name the pool
  - future release branches will now automatically switch to the -Svc pools

### CODE REVIEW

File: .azure/pipelines/jobs/default-build.yml [hunk 1]
```diff
@@ -78,6 +78,9 @@ parameters:
   # timeout due to test hangs.
   cancelTimeoutInMinutes: 15
 
+variables:
+- template: /eng/common/templates/variables/pool-providers.yml
```
Comment (dougbu) [inline .azure/pipelines/jobs/default-build.yml:82]: One sec. This doesn't work.

**Comment by dougbu** [comment]
Alright, it's working now. For example, our quarantined-pr.yml file doesn't specify the pool at all. It uses whatever pool default-build.yml chooses. The jobs in [the `aspnetcore-quarantined-pr` build](https://dev.azure.com/dnceng-public/public/_build/results?buildId=161945&view=results) show that `$(DncEngPublicBuildPool)` expands to `NetCore-Public` correctly. I'll check the expanded YAML to confirm we're not getting a default from somewhere else once that's available from one of these builds.

**Comment by dougbu** [comment]
Expanded YAML looks good. There's no apparent boundary preventing the default-build.yml expansions from seeing variables set at the top level. Transparency in this area is lightly documented but seems like something we rely on extensively.

**Review by MattGal — APPROVED** [review]
LGTM, just make sure you check out the yamls that don't run as part of PR validation after merging...

**Pull request merged**